hathor/p2p/sync_v2/transaction_streaming_client.py

```diff
@@ -37,12 +38,31 @@
 logger = get_logger()
 
 
+@dataclass
+class TransactionStreamingClientBuilder:
```

Comment:
The builder seems to be a too complex solution for the issue. I feel that it should be handled by the agent instead because it must align the request to open the stream with the client configuration. And, as a race condition might always occur between the request and the actual stream, maybe we can just remove the assert for now.

Reply:
The builder is not intended to be a solution for the issue itself. It is used as a better pattern for doing work during initialization (which basically is just calling a private method) and avoid doing it during `__init__`. It would be even an even better fit if the work was async.

But I understand your point about making the agent more aware of this. I'll leave it to just removing the assert for now then. There would be a lot of repeated logic if both the agent and the streamer client implement the logic of advancing over the list of blocks and preparing which transactions need to be downloaded if the agent has to do that before instantiating the streamer client.

Maybe an improved solution would move the implementation of that logic outside the streaming client and into a common structure that the agent uses and would be passed to the streaming client.

Reply:
Done. Assert removed.

Reply:
I agree. We can refactor it between now and the official release of sync-v2. We might add a `start()` method that would do whatever is needed without the builder.